lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/VerifyLightGBMCommon.scala

```diff
@@ -74,4 +91,213 @@ class VerifyLightGBMCommon extends TestBase {
       transposedArray.delete()
     }
   }
+
+  test("Verify sample data creation") {
+    LightGBMUtils.initializeNativeLibrary()
+    val Array(train, _) = pimaDF.randomSplit(Array(0.8, 0.2), seed)
+
+    val numRows = 100
+    val sampledRowData = train.take(numRows)
+    val featureData = sampledRowData(0).getAs[Any](featuresCol)
+    val numCols = featureData match {
+      case sparse: SparseVector => sparse.size
+      case dense: DenseVector => dense.size
+      case _ => throw new IllegalArgumentException("Unknown row data type to push")
+    }
+
+    val sampledData: SampledData = new SampledData(sampledRowData.length, numCols)
+    sampledRowData.zipWithIndex.foreach(rowWithIndex =>
+      sampledData.pushRow(rowWithIndex._1, rowWithIndex._2, featuresCol))
+
+    val rowCounts: IntSwigArray = sampledData.rowCounts
+    (0 until numCols).foreach(col => {
+      val rowCount = rowCounts.getItem(col)
+      println(s"Row counts for col $col: $rowCount")
+      val values = sampledData.sampleData.getItem(col)
+      val indexes = sampledData.sampleIndexes.getItem(col)
+      (0 until rowCount).foreach(i => println(s"  Index: ${indexes.getItem(i)}, val: ${values.getItem(i)}"))
+    })
+
+    var datasetVoidPtr:SWIGTYPE_p_p_void = null
+    try {
+      println("Creating dataset")
+      datasetVoidPtr = lightgbmlib.voidpp_handle()
+      val resultCode = lightgbmlib.LGBM_DatasetCreateFromSampledColumn(
+        sampledData.getSampleData,
+        sampledData.getSampleIndices,
+        numCols,
+        sampledData.getRowCounts,
+        numRows,
+        numRows,
+        s"max_bin=255 bin_construct_sample_cnt=$numRows min_data_in_leaf=1 num_threads=3",
+        datasetVoidPtr)
```

Comment:
This looks a little low level, is this what is expected of a user or more just for testing internal bits

Reply:
not sure what you mean.  this is a unit test of the class.  what does a user have to do with it?

Reply:
IMHO, there aren't enough "low level" tests that are really testing the code :)  We should really have more true unit tests, and not just functional API tests